aiohttp/web_protocol.py

```diff
@@ -524,22 +524,34 @@ def handle_error(self,
         self.log_exception("Error handling request", exc_info=exc)
 
         if status == 500:
-            msg = "<h1>500 Internal Server Error</h1>"
-            if self.debug:
-                with suppress(Exception):
-                    tb = traceback.format_exc()
-                    tb = html_escape(tb)
-                    msg += '<br><h2>Traceback:</h2>\n<pre>'
-                    msg += tb
-                    msg += '</pre>'
+            if 'text/html' in request.headers.get('Accept', ''):
+                if self.debug:
+                    with suppress(Exception):
+                        tb = traceback.format_exc()
+                        tb = html_escape(tb)
+                        msg = '<h2>Traceback:</h2>\n<pre>{}</pre>'.format(tb)
+                else:
+                    msg = 'Server got itself in trouble'
+                msg = (
+                    "<html><head>"
+                    "<title>500 Internal Server Error</title>"
+                    "</head><body>\n<h1>500 Internal Server Error</h1>"
+                    "<br>\n{msg}\n</body></html>\n"
+                ).format(msg=msg)
+                resp = Response(status=status, text=msg,
+                                content_type='text/html')
             else:
-                msg += "Server got itself in trouble"
-                msg = ("<html><head><title>500 Internal Server Error</title>"
-                       "</head><body>" + msg + "</body></html>")
-            resp = Response(status=status, text=msg, content_type='text/html')
+                if self.debug:
+                    with suppress(Exception):
+                        msg = traceback.format_exc()
+                else:
+                    msg = 'Server got itself in trouble'
+                msg = '500 Internal Server Error\n\n' + msg
+                resp = Response(status=status, text=msg,
+                                content_type='text/plain')
         else:
             resp = Response(status=status, text=message,
-                            content_type='text/html')
+                            content_type='text/plain')
```

Comment:
u can move this line from else if replace `msg` to `message`.
Also If u create variable content_type u can remove line 542 and 550

Reply:
done.